lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -302,6 +296,8 @@ export abstract class CacheManager implements ICacheManager {
         }
 
         if (!!cacheRecord.account) {
+            // Remove ID token claims before saving account entity
+            cacheRecord.account.idTokenClaims = undefined;
```

Comment:
Why are the claims part of the object in the first place if we're going to remove it?

On a different note, changing the value of object properties here changes it everywhere else as well even outside the context of this function. If we need to do this it's best to spread the account into a new object prior to reassigning the claims

Reply:
The AccountEntity.createAccount method was adding idTokenClaims to the entity. Went back and made sure that accounts passed into setAccount were always build by methods that didn't include ID token claims and made sure that accounts were returned in auth results with claims added.